pgmpy/inference/Sampling.py

```diff
@@ -0,0 +1,229 @@
+#!/usr/bin/env python3
+
+import networkx as nx
+from pandas import DataFrame
+from pgmpy.inference import Inference
+from pgmpy.models import BayesianModel
+from pgmpy.utils.mathext import sample_discrete
+
+
+class BayesianModelSampling(Inference):
+    """
+    Class for sampling methods specific to Bayesian Models
+
+    Parameters
+    ----------
+    model: instance of BayesianModel
+        model on which inference queries will be computed
+
+
+    Public Methods
+    --------------
+    forward_sample(size)
+    """
+    def __init__(self, model):
+        if not isinstance(model, BayesianModel):
+            raise TypeError("model must an instance of BayesianModel")
+        super(BayesianModelSampling, self).__init__(model)
+        self.topological_order = nx.topological_sort(model)
+        self.cpds = {}
+        for node in model.nodes():
+            self.cpds[node] = model.get_cpds(node)
```

Comment:
@pratyakshs You can simply do `self.cpds = {node: model.get_cpds(node) for node in model.nodes()}`